usr/src/uts/armv8/os/psci.c

```diff
@@ -120,32 +122,34 @@ psci_call(uint64_t a0, uint64_t a1, uint64_t a2, uint64_t a3)
 }
 
 void
-psci_init(void)
+psci_init(struct xboot_info *xbp)
 {
-	const struct psciinfo *pi;
+	if (psci_initialized == B_TRUE)
+		return;
 
-	pi = psciinfo_get();
-	VERIFY(pi != NULL);
+	if (xbp == NULL)
+		return;
 
-	if (pi->pi_version == PSCI_NOT_IMPLEMENTED) {
-		prom_printf("WARNING: PSCI not implemented\n");
-		psci_initialized = B_FALSE;
+	if (xbp->bi_psci_version & 0x80000000)
 		return;
-	}
 
-	switch (pi->pi_conduit) {
-	case PSCI_CONDUIT_SMC:
-		pcsi_method_is_hvc = B_FALSE;
-		break;
-	case PSCI_CONDUIT_HVC:
+	if (xbp->bi_psci_conduit == BI_PSCI_CONDUIT_HVC)
 		pcsi_method_is_hvc = B_TRUE;
-		break;
-	}
+	else if (xbp->bi_psci_conduit == BI_PSCI_CONDUIT_SMC)
+		pcsi_method_is_hvc = B_FALSE;
+	else
+		return;
 
-	psci_cpu_suspend_id = pi->pi_cpu_suspend_id;
-	psci_cpu_off_id = pi->pi_cpu_off_id;
-	psci_cpu_on_id = pi->pi_cpu_on_id;
-	psci_migrate_id = pi->pi_migrate_id;
+	if (((xbp->bi_psci_version & 0x7FFF0000) >> 16) == 0) {
+		if (xbp->bi_psci_cpu_suspend_id != 0)
+			psci_cpu_suspend_id = xbp->bi_psci_cpu_suspend_id;
+		if (xbp->bi_psci_cpu_off_id != 0)
+			psci_cpu_off_id = xbp->bi_psci_cpu_off_id;
+		if (xbp->bi_psci_cpu_on_id != 0)
+			psci_cpu_on_id = xbp->bi_psci_cpu_on_id;
+		if (xbp->bi_psci_migrate_id != 0)
+			psci_migrate_id = xbp->bi_psci_migrate_id;
+	}
```

Comment:
Similar question, do we want to print something if this condition fails?

Reply:
These are a little different. The ability to override the function IDs is only relevant in PSCI 0.2 and the prior draft. From PSCI 1.0 the function identifiers are fixed.